### PR TITLE
Added octavia-operator pre-commit job

### PR DESCRIPTION
**Pull request by amolkahat**
This patch adds pre-commit job for octavia
operator. This job is optional for now.

### CODE REVIEW

**Comment by amolkahat** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-octavia-operator-main-precommit-check

**Comment by amolkahat** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-octavia-operator-main-precommit-check

**Comment by amolkahat** [comment]
/pj-rehearse pull-ci-openstack-k8s-operators-octavia-operator-main-precommit-check

**Comment by Sandeepyadav93** [comment]
Fix: https://github.com/openstack-k8s-operators/octavia-operator/pull/103 should merge first, before this pr.

File: ci-operator/config/openstack-k8s-operators/octavia-operator/openstack-k8s-operators-octavia-operator-main.yaml [hunk 1]
```diff
@@ -97,6 +102,16 @@ tests:
...
   steps:
     workflow: openstack-k8s-operators-test-build-deploy-kuttl
+- as: precommit-check
+  commands: |
+    export SKIP=no-commit-to-branch
+    export GOFLAGS=
+    export PRE_COMMIT_HOME=/tmp
+    export GOLANGCI_LINT_CACHE=/tmp
+    mkdir -p ../operator && cp -r . ../operator
+    pre-commit run -a
```
Comment (Sandeepyadav93) [inline ci-operator/config/openstack-k8s-operators/octavia-operator/openstack-k8s-operators-octavia-operator-main.yaml:112]: "pre-commit run --all-files --show-diff-on-failure"

Please sync this with https://github.com/openshift/release/blob/master/ci-operator/config/openstack-k8s-operators/openstack-operator/openstack-k8s-operators-openstack-operator-master.yaml#L75-L86

**Pull request merged**